src/librustc/infer/error_reporting/anon_anon_conflict.rs

```diff
@@ -27,66 +27,97 @@ impl<'a, 'gcx, 'tcx> InferCtxt<'a, 'gcx, 'tcx> {
     //    { x.push(y); }.
     // The example gives
     // fn foo(x: &mut Vec<&u8>, y: &u8) {
-    //                    ---      --- these references must have the same lifetime
+    //                    ---      --- these references are declared with different lifetimes...
     //            x.push(y);
-    //            ^ data from `y` flows into `x` here
-    // It will later be extended to trait objects and structs.
+    //            ^ ...but data from `y` flows into `x` here
+    // It has been extended for the case of structs too.
+    // Consider the example
+    // struct Ref<'a> { x: &'a u32 }
+    // fn foo(mut x: Vec<Ref>, y: Ref) {
+    //                   ---      --- these structs are declared with different lifetimes...
+    //               x.push(y);
+    //               ^ ...but data from `y` flows into `x` here
+    // }
+    // It will later be extended to trait objects.
     pub fn try_report_anon_anon_conflict(&self, error: &RegionResolutionError<'tcx>) -> bool {
-
         let (span, sub, sup) = match *error {
             ConcreteFailure(ref origin, sub, sup) => (origin.span(), sub, sup),
             _ => return false, // inapplicable
         };
 
         // Determine whether the sub and sup consist of both anonymous (elided) regions.
-        let (ty1, ty2) = if self.is_suitable_anonymous_region(sup).is_some() &&
-                            self.is_suitable_anonymous_region(sub).is_some() {
+        let (ty1, ty2, scope_def_id_1, scope_def_id_2, bregion1, bregion2) = if
+            self.is_suitable_anonymous_region(sup, true).is_some() &&
```

Comment:
Another possible syntax that could make you life better:

```Rust
let sup_reg = match self.is_suitable_anonymous_region(sup, true) {
    Some(val) => val,
    None => return false
};
```

Or you could use `Result<whatever, ReportFailed>` for your own `struct ReportFailed;` and do:
```Rust
let sup_reg = self.is_suitable_anonymous_region(sup, true)?;
```